src/aten/Distributions.cpp

```diff
@@ -0,0 +1,221 @@
+          TORCH_CHECK(
+              from <= to,
+              "uniform_ expects to return a [from, to) range, but found from=",
+              from,
+              " > to=",
+              to);
+          TORCH_CHECK(
+              (to - from) <= std::numeric_limits<scalar_t>::max(),
+              "uniform_ expects to-from <= std::numeric_limits<",
+              toString(self.scalar_type()),
+              ">::max(), but found to=",
+              to,
+              " and from=",
+              from,
+              " which result in to-from to exceed the limit");
+          from = std::min(std::max(from, min), max);
+          to = std::max(std::min(to, max), min);
+        });
+    CHECK_EMPTY_AND_RETURN(self);
+    auto iter = at::TensorIterator::borrowing_nullary_op(self);
+    native::xpu::uniform_kernel(iter, from, to, generator);
+  }
+  return self;
+}
+
+Tensor& XPUNativeFunctions::bernoulli_(
+    Tensor& self,
+    const Tensor& p_,
+    ::std::optional<Generator> generator) {
+  native::xpu::bernoulli_tensor_kernel(self, p_, std::move(generator));
+  return self;
+}
+
+Tensor& XPUNativeFunctions::bernoulli_(
+    Tensor& self,
+    double p,
+    ::std::optional<Generator> generator) {
+  native::xpu::bernoulli_scalar_kernel(self, p, std::move(generator));
+  return self;
+}
+
+#define CHECK_OUT_OF_BOUNDS(var, name, min, max, dtype) \
+  TORCH_CHECK(var >= min && var <= max, name, " is out of bounds for ", dtype);
+
+#define WARN_OUT_OF_BOUNDS(var, name, digits, dtype)                                                 \
+  if (var < -(1LL << digits) || var > (1LL << digits)) {                                             \
+    TORCH_WARN(                                                                                      \
+        name,                                                                                        \
+        " is out of bounds [-(2^",                                                                   \
+        digits,                                                                                      \
+        "), 2^",                                                                                     \
+        digits,                                                                                      \
+        "]. ",                                                                                       \
+        "Due to precision limitations ",                                                             \
+        dtype,                                                                                       \
+        " can support discrete uniform distribution only within this range. ",                       \
+        "This warning will become an error in version 1.7 release, please fix the code in advance"); \
+  }
+
+Tensor& XPUNativeFunctions::random_(
+    Tensor& self,
+    int64_t from,
+    c10::optional<int64_t> to_opt,
+    ::std::optional<Generator> generator) {
+  uint64_t range = 0;
```

Comment:
Can we share the whole template? We can pass random_kernel and RNG as arguments of the template?
<img width="773" alt="image" src="https://github.com/intel/torch-xpu-ops/assets/28250760/3790723a-12c0-4d86-83c8-99a15d76d3ae">

Reply:
Then we are simple here.